src/test_runner.cr

```diff
@@ -45,16 +45,46 @@ module Mint
 
                 this.suites = suites
 
+                let error = null;
+
+                window.onerror = (message) => {
+                  if (this.socket.readyState === 1) {
+                    console.log("!!1");
+                    this.crash(message);
+                  } else {
+                    console.log("!!2");
+                    error ||= message;
+                  }
+                }
+
                 this.socket.onopen = () => {
-                  this.run()
-                    .then(() => this.socket.send("DONE"))
+                  if (error != null) {
+                    console.log("!!3");
+                    this.crash(error);
+                  }
+
+                  window.addEventListener('unhandledrejection', (event) => {
+                    event.promise.catch(e => crash(e.message));
+                  });
+                }
+              }
+
+              start (suites) {
+                if (this.socket.readState === 1) {
+                  this.run();
+                } else {
+                  this.socket.addEventListener("open", this.run);
                 }
               }
 
-              async run () {
+              run () {
                 return new Promise((resolve, reject) => {
                   this.next(resolve, reject)
-                })
+                }).then(() => this.socket.send("DONE"));
```

Comment:
```suggestion
                }).finally(() => this.socket.send("DONE"));
```

Reply:
Ah, good idea!

Reply:
I've noticed you've changed it back to `then`, why?